sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/InsertTableWithDynamicPartitionsBenchmark.scala

```diff
@@ -0,0 +1,138 @@
+
+  override def getSparkSession: SparkSession = {
+    SparkSession.builder().master("local[4]").getOrCreate()
+  }
+
+  def prepareSourceTableAndGetTotalRowsCount(numberRows: Long,
+      part1Step: Int, part2Step: Int, part3Step: Int): Long = {
+    val dataFrame = spark.range(0, numberRows, 1, 4)
+    val dataFrame1 = spark.range(0, numberRows, part1Step, 4)
+    val dataFrame2 = spark.range(0, numberRows, part2Step, 4)
+    val dataFrame3 = spark.range(0, numberRows, part3Step, 4)
+
+    val data = dataFrame.join(dataFrame1).join(dataFrame2).join(dataFrame3)
+      .toDF("id", "part1", "part2", "part3")
+
+    data.createOrReplaceTempView("tmpTable")
+
+    spark.sql("create table " +
+      "sourceTable(id bigint, part1 bigint, part2 bigint, part3 bigint) " +
+      "using parquet")
+
+    spark.sql("insert overwrite table sourceTable " +
+      s"select id, " +
+      s"part1, " +
+      s"part2, " +
+      s"part3 " +
+      s"from tmpTable")
+
+    spark.catalog.dropTempView("tmpTable")
+    data.count()
+  }
+
+  def prepareTable(): Unit = {
+    spark.sql("create table " +
+      "tableOnePartitionColumn(i bigint, part bigint) " +
+      "using parquet partitioned by (part)")
+    spark.sql("create table " +
+      "tableTwoPartitionColumn(i bigint, part1 bigint, part2 bigint) " +
+      "using parquet partitioned by (part1, part2)")
+    spark.sql("create table " +
+      "tableThreePartitionColumn(i bigint, part1 bigint, part2 bigint, part3 bigint) " +
+      "using parquet partitioned by (part1, part2, part3)")
+  }
+
+  def writeOnePartitionColumnTable(partitionNumber: Long, benchmark: Benchmark): Unit = {
+    benchmark.addCase(s"one partition column, $partitionNumber partitions") { _ =>
+      spark.sql("insert overwrite table " +
+        "tableOnePartitionColumn partition(part) " +
+        s"select id, " +
+        s"part1 as part " +
+        s"from sourceTable")
+    }
+  }
+
+  def writeTwoPartitionColumnTable(partitionNumber: Long, benchmark: Benchmark): Unit = {
+    benchmark.addCase(s"two partition columns, $partitionNumber partitions") { _ =>
+      spark.sql("insert overwrite table " +
+        "tableTwoPartitionColumn partition(part1, part2) " +
+        s"select id, " +
+        s"part1, " +
+        s"part2 " +
+        s"from sourceTable")
+    }
+  }
+
+  def writeThreePartitionColumnTable(partitionNumber: Long, benchmark: Benchmark): Unit = {
+    benchmark.addCase(s"three partition columns, $partitionNumber partitions") { _ =>
+      spark.sql("insert overwrite table " +
+        "tableThreePartitionColumn partition(part1, part2, part3) " +
+        s"select id, " +
+        s"part1, " +
+        s"part2, " +
+        s"part3 " +
+        s"from sourceTable")
+    }
+  }
+
+  override def runBenchmarkSuite(mainArgs: Array[String]): Unit = {
+    val sourceTable = "sourceTable"
+    val tableOnePartitionColumn = "tableOnePartitionColumn"
```

Comment:
it can be simpler `val onePartTable =  ...`

Reply:
done ~